CiviEvent - Fix iCalendar regression (RC)

totten opened the pull request:
Forward-port #28702 from 5.68-stable to 5.69-rc

MegaphoneJon committed:
Fix Event iCalendar regression

eileenmcnaughton commented:
Adding MOP for completeness - in fact this is just a port so @totten can self-merge

totten commented:
Did some `r-run` -- reproduced the bug and confirmed the patch fixes it. 👍

totten committed:
CiviEvent - Re-fix iCal. Preserve compatibility.

The recently merged commit for `CRM_Utils_ICalendar` fixed one bug and did
some code-cleanup...  but it changed a contract in a way that breaks
`eventics` extension.

This restores the original contract and a more conservative fix.

totten commented:
This had MOP, but I dropped pending another revision. That revision was subsequently written / pushed / passed. Multiple 👍s on MM. Merging.
